### DISCUSSION TITLE
How github-tags  work with on-prem github entreprise server ?

### DISCUSSION THREAD

**Discussion by NargiT** [renovatebot/renovate#16461] [Request Help] [answered]
Hello,
Hello,

I would like to use regexManagers with github-tags. The goal is to use upgrade values inside a cutsom yaml file and use local github tags or releases as a registry. 

**Context**:
- Github Entreprise Server (On-Prem)
- renovate as a self-hosted github action

```json
  "regexManagers": [
    {
      "fileMatch": [
        "(^|/)config.yaml$"
      ],
      "matchStrings": [
        "cloud_template:[\\s\\w:\\-]+name:\\s*(?<depName>.*?)\\s*?hash:\\s*(?<currentValue>.*?)\n"
      ],
      "datasourceTemplate": "github-tags",
      "versioningTemplate": "semver",
      "registryUrlTemplate": "https://github.mycompany.com/ORG-A/"
    }
```

I also added a entry in hostRules.

```json
"hostRules": [
 {
  "hostType": "github",
  "matchHost": "github.mycompany.com/",
  "authType": "Token-Only",
  "token": "ghp_token" <-- later I would like to use github app token but for the purpose of the test I am using a PAT
 }
]
```

This is the file I try to upgrade

```yaml
cloud_template:
  name: my-dep
  hash: v1.0.0
```

And when I run renovate in my CI with LOG_LEVEL=debug I see this information in logs.

```log
DEBUG: Found no results from datasource that look like a version (repository=ORGA-A/test-renovate, dependency=my-dep)
       "result": {
         "sourceUrl": "https://github.mycompany.com/ORG-A/my-dep",
         "releases": []
       }
```

Any help or pointers are appreciated. So far I only found this documentation for [github-tags](https://docs.renovatebot.com/modules/datasource/#github-tags-datasource) but it's very light.

**Comment by viceice** [comment, marked as answer]
use depNameTemplate and prepend the org to `depName`. remove org from registry url.

**Reply by viceice** [reply]
or use package name template, so depName stays short.

**Reply by NargiT** [reply]
I will try and post the result if it does not work.

**Reply by NargiT** [reply]
here is the final config, kudos to @viceice 

```json
  "regexManagers": [
    {
      "fileMatch": [
        "(^|/)config.yaml$"
      ],
      "matchStrings": [
        "cloud_template:[\\s\\w:\\-]+name:\\s*(?<depName>.*?)\\s*?hash:\\s*(?<currentValue>.*?)\n"
      ],
      "datasourceTemplate": "github-tags",
      "versioningTemplate": "semver",
      "packageNameTempalte": "ORG-A/{{{depName}}}"
      "registryUrlTemplate": "https://github.mycompany.com/"
    }
```
and the correct hostrules, because in our case the whole GHE is private
```json
"hostRules": [
 {
  "hostType": "github",
  "matchHost": "https://github.mycompany.com/",
  "authType": "Token-Only",
  "token": "ghp_token"
 }
]
```


side note: if you use a github app with github action, you don't need to define the token in host rules.
